integrated_channels/moodle/models.py

```diff
@@ -106,8 +109,92 @@ def channel_code():
         """
         return 'MOODLE'
 
+    def get_learner_data_exporter(self, user):
+        return MoodleLearnerExporter(user, self)
+
+    def get_learner_data_transmitter(self):
+        return MoodleLearnerTransmitter(self)
+
     def get_content_metadata_exporter(self, user):
         return MoodleContentMetadataExporter(user, self)
 
     def get_content_metadata_transmitter(self):
         return MoodleContentMetadataTransmitter(self)
+
+
+@python_2_unicode_compatible
+class MoodleLearnerDataTransmissionAudit(models.Model):
+    """
+    The payload we send to Moodle at a given point in time for an enterprise course enrollment.
+
+    """
+    moodle_user_email = models.CharField(max_length=255, blank=False, null=False)
+
+    enterprise_course_enrollment_id = models.PositiveIntegerField(blank=False, null=False, db_index=True)
+    course_id = models.CharField(max_length=255, blank=False, null=False)
+    course_completed = models.BooleanField(default=True)
+    grade = models.CharField(max_length=100, blank=False, null=False)
+    total_hours = models.FloatField(null=True, blank=True)
+    course_completed = models.BooleanField(
+        default=True,
```

Comment:
minor question I guess, is default True sensible because we are already transmitting grades? is default False any better/worse?

Reply:
Yeah, I have the default here because we have other checks in the transmission flow that make sure the course is completed. I don't think it's necessary either way, but willing to make a change if you think it's better practice to "assume it's not until we prove it is"

Reply:
based on how your code is setup this change is not needed, but yeah as a general dataa modeling concern perhaps it makes sense to default to False? then someone can explicitly set to True if conditions are right. I think a False for this value will mean no grade transmission, so wil leave it up to you